// @smileid/web-components/dist/esm/package-B-UwEdv7.js
const n = {
  /** ID_CARD_BACK_IMAGE_BASE64 Base64 encoded back of ID card image (.jpg or .png) */
  ID_CARD_BACK_IMAGE_BASE64: 7,
  /** ID_CARD_BACK_IMAGE_FILE Back of ID card image in .png or .jpg file format */
  ID_CARD_BACK_IMAGE_FILE: 5,
  /** ID_CARD_IMAGE_BASE64 Base64 encoded ID card image (.png or .jpg) */
  ID_CARD_IMAGE_BASE64: 3,
  /** ID_CARD_IMAGE_FILE ID card image in .png or .jpg file format */
  ID_CARD_IMAGE_FILE: 1,
  /** LIVENESS_IMAGE_BASE64 Base64 encoded liveness image (.jpg or .png) */
  LIVENESS_IMAGE_BASE64: 6,
  /** LIVENESS_IMAGE_FILE Liveness image in .png or .jpg file format */
  LIVENESS_IMAGE_FILE: 4,
  /** SELFIE_IMAGE_BASE64 Base64 encoded selfie image (.png or .jpg) */
  SELFIE_IMAGE_BASE64: 2,
  /** SELFIE_IMAGE_FILE Selfie image in .png or .jpg file format */
  SELFIE_IMAGE_FILE: 0
};
class a {
  static stream = null;
  static async getMedia(e) {
    return a.stream = await navigator.mediaDevices.getUserMedia(e), a.stream;
  }
  static environmentOptions = {
    facingMode: "environment",
    height: {
      ideal: 1080,
      max: 1440,
      min: 720
    },
    width: {
      ideal: 1920,
      max: 2560,
      min: 1280
    },
    // NOTE: Special case for multi-camera Samsung devices (learnt from Acuant)
    // "We found out that some triple camera Samsung devices (S10, S20, Note 20, etc) capture images blurry at edges.
    // Zooming to 2X, matching the telephoto lens, doesn't solve it completely but mitigates it."
    zoom: a.isSamsungMultiCameraDevice() ? 2 : 1
  };
  static stopMedia() {
    a.stream && (a.stream.getTracks().forEach((e) => e.stop()), a.stream = null);
  }
  static async supportsAgentMode() {
    try {
      const r = (await navigator.mediaDevices.enumerateDevices()).filter(
        (t) => t.kind === "videoinput"
      );
      let s = !1;
      return r.forEach((t) => t.label.toLowerCase().includes("back") || t.label.toLowerCase().includes("rear") ? (s = !0, !0) : !1), s;
    } catch (e) {
      return console.warn("Error accessing media devices: ", e), !1;
    }
  }
  static isSamsungMultiCameraDevice() {
    const e = navigator.userAgent.match(/SM-[N|G]\d{3}/);
    if (!e)
      return !1;
    const r = parseInt(e[0].match(/\d{3}/)[0], 10);
    return !Number.isNaN(r) && r >= 970;
  }
  static handleCameraError(e) {
    switch (e.name) {
      case "NotAllowedError":
      case "SecurityError":
        return `
              Looks like camera access was not granted, or was blocked by a browser
              level setting / extension. Please follow the prompt from the URL bar,
              or extensions, and enable access.
              You may need to refresh to start all over again
            `;
      case "AbortError":
        return `
              Oops! Something happened, and we lost access to your stream.
              Please refresh to start all over again
            `;
      case "NotReadableError":
        return `
              There seems to be a problem with your device's camera, or its connection.
              Please check this, and when resolved, try again. Or try another device.
            `;
      case "NotFoundError":
        return `
              We are unable to find a video stream.
              You may need to refresh to start all over again
            `;
      case "TypeError":
        return `
              This site is insecure, and as such cannot have access to your camera.
              Try to navigate to a secure version of this page, or contact the owner.
            `;
      default:
        return e.message;
    }
  }
}
const o = "11.0.3", i = {
  version: o
};
export {
  n as I,
  a as S,
  i as p
};
